tests/unit/lms/resources/_js_config/__init___test.py

```diff
@@ -222,34 +223,40 @@ def test_it_sets_the_document_url(self, js_config, submission_params):
 
         assert submission_params()["document_url"] == "example_document_url"
 
+    @pytest.mark.parametrize("submit_on_annotation_enabled", [True, False])
     def test_it_sets_the_canvas_submission_params(
-        self, pyramid_request, submission_params, js_config
+        self,
+        pyramid_request,
+        js_config,
+        submission_params,
+        submit_on_annotation_enabled,
     ):
+        if submit_on_annotation_enabled:
+            pyramid_request.feature.side_effect = (
+                lambda feature: feature == "submit_on_annotation"
+            )
 
         js_config.add_document_url("canvas://file/course_id/COURSE_ID/file_id/FILE_ID")
 
-        assert (
-            submission_params()["h_username"]
-            == pyramid_request.lti_user.h_user.username
-        )
-        assert (
-            submission_params()["lis_outcome_service_url"]
-            == "example_lis_outcome_service_url"
-        )
-        assert (
-            submission_params()["lis_result_sourcedid"]
-            == "example_lis_result_sourcedid"
+        assert submission_params() == Any.dict.containing(
+            {
+                "h_username": pyramid_request.lti_user.h_user.username,
+                "lis_outcome_service_url": "example_lis_outcome_service_url",
+                "lis_result_sourcedid": "example_lis_result_sourcedid",
+                "learner_canvas_user_id": "test_user_id",
+                "resource_link_id": pyramid_request.params["resource_link_id"],
+                "ext_lti_assignment_id": pyramid_request.params[
+                    "ext_lti_assignment_id"
+                ],
+            }
         )
-        assert submission_params()["learner_canvas_user_id"] == "test_user_id"
 
-        assert (
-            submission_params()["resource_link_id"]
-            == pyramid_request.params["resource_link_id"]
-        )
-        assert (
-            submission_params()["ext_lti_assignment_id"]
-            == pyramid_request.params["ext_lti_assignment_id"]
-        )
+        if submit_on_annotation_enabled:
+            # pylint:disable=protected-access
+            assert js_config._hypothesis_client["reportActivity"] == {
```

Comment:
The fixture `config` won't have `hypothesisClient` unless `enable_launch_mode` is called. I think accessing the private property directly is less confusing than calling enable_launch_mode for its side effects.